src/test/bridges/angle/AngleSLPE2E.t.sol

```diff
@@ -0,0 +1,211 @@
+import {SafeERC20} from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import {ICERC20} from "../../../interfaces/compound/ICERC20.sol";
+import {AngleSLPBridge} from "../../../bridges/angle/AngleSLPBridge.sol";
+import {ErrorLib} from "../../../bridges/base/ErrorLib.sol";
+
+contract AngleSLPE2ETest is BridgeTestBase {
+    using SafeERC20 for IERC20;
+
+    AztecTypes.AztecAsset private ethAsset;
+
+    AztecTypes.AztecAsset private daiAsset;
+    AztecTypes.AztecAsset private usdcAsset;
+    AztecTypes.AztecAsset private wethAsset;
+
+    AztecTypes.AztecAsset private sanDaiAsset;
+    AztecTypes.AztecAsset private sanUsdcAsset;
+    AztecTypes.AztecAsset private sanWethAsset;
+
+    AngleSLPBridge internal bridge;
+    uint256 internal id;
+
+    uint256 internal constant DUST = 1;
+
+    function setUp() public {
+        emit log_named_uint("block number", block.number);
+        bridge = new AngleSLPBridge(address(ROLLUP_PROCESSOR));
+        vm.label(address(bridge), "ANGLE_BRIDGE");
+        vm.deal(address(bridge), 0);
+
+        vm.startPrank(MULTI_SIG);
+        ROLLUP_PROCESSOR.setSupportedBridge(address(bridge), 200000);
+        ROLLUP_PROCESSOR.setSupportedAsset(0x7B8E89b0cE7BAC2cfEC92A371Da899eA8CBdb450, 100000); // sanDAI
+        ROLLUP_PROCESSOR.setSupportedAsset(0xA0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48, 100000); // USDC
+        ROLLUP_PROCESSOR.setSupportedAsset(0x9C215206Da4bf108aE5aEEf9dA7caD3352A36Dad, 100000); // sanUSDC
+        ROLLUP_PROCESSOR.setSupportedAsset(0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2, 100000); // weth
+        ROLLUP_PROCESSOR.setSupportedAsset(0x30c955906735e48D73080fD20CB488518A6333C8, 100000); // sanWETH
+        vm.stopPrank();
+
+        id = ROLLUP_PROCESSOR.getSupportedBridgesLength();
+
+        ethAsset = getRealAztecAsset(address(0));
+
+        daiAsset = getRealAztecAsset(0x6B175474E89094C44Da98b954EedeAC495271d0F);
+        sanDaiAsset = getRealAztecAsset(0x7B8E89b0cE7BAC2cfEC92A371Da899eA8CBdb450);
+
+        usdcAsset = getRealAztecAsset(0xA0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48);
+        sanUsdcAsset = getRealAztecAsset(0x9C215206Da4bf108aE5aEEf9dA7caD3352A36Dad);
+
+        wethAsset = getRealAztecAsset(0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2);
+        sanWethAsset = getRealAztecAsset(0x30c955906735e48D73080fD20CB488518A6333C8);
+
+        deal(daiAsset.erc20Address, address(bridge), DUST);
+        deal(sanDaiAsset.erc20Address, address(bridge), DUST);
+        deal(usdcAsset.erc20Address, address(bridge), DUST);
+        deal(sanUsdcAsset.erc20Address, address(bridge), DUST);
+        deal(wethAsset.erc20Address, address(bridge), DUST);
+        deal(sanWethAsset.erc20Address, address(bridge), DUST);
+
+        // since the rollup doesn't have USDC, we deal some to it
+        deal(usdcAsset.erc20Address, address(ROLLUP_PROCESSOR), 100000e6);
+    }
+
+    function testValidDepositE2E(uint96 _amount) public {
```

Comment:
This and the USDC test are essentially the same test, just with different input/output assets. You might want to have `_testValidDepositE2E(_inputAsset, _outputAsset, _amount)` instead, and then call that with the different assets to not need the similar tests.